code/controllers/subsystem/sound.dm

```diff
@@ -0,0 +1,118 @@
+/*
+ * Sound subsystem:
+ * Used for things that need constant updating (sound fading in / out)
+*/
+
+SUBSYSTEM_DEF(sound_effects)
+	name = "Sound"
+	wait = 1
+	priority = FIRE_PRIORITY_AMBIENCE
+	flags = SS_NO_INIT
+	//Note: Make sure you update this if you use sound fading pre-game
+	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	var/list/acting_effects = list()	//key = sound, value = datum
+	var/list/currentrun = list()
+
+/datum/controller/subsystem/sound_effects/fire(resumed = 0)
+	if (!resumed)
+		src.currentrun = acting_effects.Copy()
+
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+
+	while(LAZYLEN(currentrun))
+		var/datum/sound_effect/sound_effect = currentrun[currentrun[currentrun.len]]
+		currentrun.len--
+
+		sound_effect.update_effect()
+
+		if(world.time > sound_effect.end_tick)
+			sound_effect.end_effect()
+			acting_effects -= sound_effect.effect_id
+
+		if (MC_TICK_CHECK)
+			return
+
+// ===== Sound effect procs =====
+
+/proc/sound_fade(sound/S, start_volume = 100, end_volume = 0, time = 10, var/listeners)
```

Comment:
I don't like this. This should all be done in the New() of /sound_effect and the New() of /sound_effect/fade

Possibly add a proc to sound effects to begin them